src/node/nodestate.h

```diff
@@ -1162,6 +1096,121 @@ namespace ccf
       secrets_view->put(version, secret_set);
     }
 
+    KeyShareInfo generate_key_share_info(
+      const std::vector<ccf::MemberPubInfo>& members_info)
+    {
+      auto share_wrapping_key_raw =
+        tls::create_entropy()->random(crypto::GCM_SIZE_KEY);
+      auto share_wrapping_key = crypto::KeyAesGcm(share_wrapping_key_raw);
+
+      // Encrypt initial ledger secrets with k_z
+      // Once sealing is completely removed, this can be called to the
+      // LedgerSecrets class directly
+      crypto::GcmCipher encrypted_ls(
+        network.ledger_secrets->get_secret(1)->master.size());
+
+      share_wrapping_key.encrypt(
+        encrypted_ls.hdr.get_iv(), // iv is always 0 here as the share wrapping
+                                   // key is never re-used for encryption
+        network.ledger_secrets->get_secret(1)->master,
+        nullb,
+        encrypted_ls.cipher.data(),
+        encrypted_ls.hdr.tag);
+
+      auto active_members = members_info.size();
+
+      // For now, the secret sharing threashold is set to the number of initial
+      // members
+      size_t k = active_members;
```

Comment:
number_of_shares is perhaps more explicit than k?

Reply:
`number_of_shares` is more appropriate for `n` than for `k`. Let's use `threshold` for now. This code is going to be revisited very soon anyway.